redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,493 @@
+    }
+
+    asyncResp->res.jsonValue["Assemblies"] = nlohmann::json::array();
+    asyncResp->res.jsonValue["Assemblies@odata.count"] = 0;
+
+    // check if this chassis hosts any association
+    dbus::utility::getDbusObject(
+        chassisPath, std::array<std::string_view, 0>(),
+        [asyncResp, chassisPath](const boost::system::error_code ec,
+                                 const dbus::utility::MapperGetObject& object) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        for (const auto& [serviceName, interfaceList] : object)
+        {
+            for (const auto& interface : interfaceList)
+            {
+                if (interface == "xyz.openbmc_project.Association.Definitions")
+                {
+                    checkForAssemblyAssociations(asyncResp, chassisPath,
+                                                 serviceName);
+                    return;
+                }
+            }
+        }
+        return;
+    });
+}
+
+namespace assembly
+{
+/**
+ * @brief Get chassis path with given chassis ID
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisID - Chassis to which the assemblies are
+ * associated.
+ *
+ * @return None.
+ */
+inline void getChassis(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       const std::string& chassisID)
+{
+    BMCWEB_LOG_DEBUG("Get chassis path");
+
+    constexpr std::array<std::string_view, 2> interfaces = {
+        "xyz.openbmc_project.Inventory.Item.Chassis",
+        "xyz.openbmc_project.Inventory.Item.Board"};
+
+    // get the chassis path
+    dbus::utility::getSubTreePaths(
+        "/xyz/openbmc_project/inventory", 0, interfaces,
+        [asyncResp, chassisID](
+            const boost::system::error_code ec,
+            const dbus::utility::MapperGetSubTreePathsResponse& chassisPaths) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        // check if the chassis path belongs to the chassis ID passed
+        for (const auto& path : chassisPaths)
+        {
+            BMCWEB_LOG_DEBUG("Chassis Paths from Mapper ", path);
+            std::string chassis =
+                sdbusplus::message::object_path(path).filename();
+            if (chassis != chassisID)
+            {
+                // this is not the chassis we are interested in
+                continue;
+            }
+
+            asyncResp->res.jsonValue["@odata.type"] =
+                "#Assembly.v1_3_0.Assembly";
+            asyncResp->res.jsonValue["@odata.id"] = boost::urls::format(
+                "/redfish/v1/Chassis/{}/Assembly", chassisID);
+            asyncResp->res.jsonValue["Name"] = "Assembly Collection";
+            asyncResp->res.jsonValue["Id"] = "Assembly";
```

Comment:
Should these code block L458-L463 to be moved inside checkAssociation at L383?

Reply:
I've moved it to L383, but I don't find any difference in output, as the output JSON structures the contents in alphabetical order.

Reply:
What I'm suggesting is better to group together.

The content should not be changing in normal cases.
However, what will happen  `if (chassis.empty())`? The content may become inconsistent if not moved into checkAssociation.

Reply:
got you 👍